### DISCUSSION TITLE
GitHub workflows started to fails

### DISCUSSION THREAD

**Discussion by afbora** [community/community#27056] [Actions]
<p>Workflows pinned actions with SHA have been failing since yesterday.</p>
<blockquote>
<p>Failed to resolve action download info. Error: Internal Server Error</p>
</blockquote>
<p>Workflow Job:</p><aside class="onebox githubactions">
  <header class="source">

      <a href="https://github.com/getkirby/kirby/actions/runs/936580721" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">

      <svg width="24" height="24" class="github-icon github-icon-failure" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm8.036-4.024a.75.75 0 00-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 101.06 1.06L12 13.06l2.963 2.964a.75.75 0 001.061-1.06L13.061 12l2.963-2.964a.75.75 0 10-1.06-1.06L12 10.939 9.036 7.976z"></path></svg>

  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/getkirby/kirby/actions/runs/936580721" target="_blank" rel="noopener nofollow ugc">Fixes toggle method in files field dropdown</a>
    </h4>

      <div class="github-info">
        CI <span class="github-run-number">#2137</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>It works when I pin it with version instead of SHA of v2:</p>
<p>Works: <code>actions/checkout@v2</code><br>
Fails: <code>actions/checkout@f1d3225b5376a0791fdee5a0e8eac5289355e43a</code></p>

**Comment by airtower-luna** [comment]
<p>That commit doesn’t seem to exist in the <a href="https://github.com/actions/checkout"><code>actions/checkout</code> repository</a>. That explains the failed download, but if the same commit ID worked before it seems like something weird is going on. :thinking:</p>

**Comment by afbora** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="185188">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>That commit doesn’t seem to exist in the <a href="https://github.com/actions/checkout" rel="noopener nofollow ugc"> <code>actions/checkout</code> repository </a></p>
</blockquote>
</aside>
<p>Do you mean <code>f1d3225b5376a0791fdee5a0e8eac5289355e43a</code> SHA?</p>

**Comment by airtower-luna** [comment]
<p>Yes, exactly. I have no idea <em>why</em>, but trying to view the commit doesn’t work (404).</p>

**Comment by afbora** [comment]
<p>This is SHA of <code>v2</code> ref, not commit id. You can see via API:</p>
<p><a href="https://api.github.com/repos/actions/checkout/git/refs/tags" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/actions/checkout/git/refs/tags</a></p>

**Comment by afbora** [comment]
<p>Also, it was already working like this. It suddenly stopped working yesterday.</p>

**Comment by afbora** [comment]
<p>The issue may not be exactly about pinning the SHA either. The only error I can see is this.</p>

**Comment by airtower-luna** [comment]
<p>As I understand the documentation (<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses"><code>jobs.&lt;job_id&gt;.steps[*].uses</code></a>) specifying a commit SHA to pin the version is the intended use, not other objects. So unless pinning the commit ID fails, too, my <em>guess</em> is that there was some internal change that broke something that wasn’t intended to work.</p>

**Comment by afbora** [comment]
<p>As you said, using the latest  version commit hashes fixed the issue. But since it broke while working with tag hashes, I reported the situation to the github support team. Thanks.</p>

**Comment by joshmgross** [comment, marked as answer]
<p>:wave: Hey <a class="mention" href="/afbora">@afbora</a>,</p>
<p>This was broken by an internal change we made, sorry about that. I’ve disabled that change, so referencing a tag by the SHA object ID should work for now.</p>
<p>As referenced in our documentation, the intended use is using a commit SHA rather than other objects.</p>

**Comment by Johnvan** [comment]
<p>You use too many acronyms which I have no idea of their meaning such as Pull and here the use of SHA.</p>

**Comment by airtower-luna** [comment]
<p>To learn Git terms, take a look at the <a href="https://git-scm.com/book/en/v2">Git Book</a>. :slightly_smiling_face:</p>
